pkg/xds/context/mesh_context_builder.go

```diff
@@ -297,10 +291,23 @@ func (m *meshContextBuilder) BuildBaseMeshContextIfChanged(ctx context.Context,
 	if latest != nil && bytes.Equal(newHash, latest.hash) {
 		return latest, nil
 	}
+
+	var destinations []DestinationResource
+	for _, resources := range rmap {
+		if !isDestinationResource(resources) {
+			continue
+		}
+
+		for _, res := range resources.GetItems() {
+			destinations = append(destinations, res.(DestinationResource))
+		}
+	}
```

Comment:
This is the hotpath (we create these contexts all the time).
The destinations list you are creating here is thrown away straight after the init building an iterator may be a better option.

Also using type descriptor after this code could be part of:
https://github.com/kumahq/kuma/blob/ce3342fe9e4489f9f42259f1e19d8b294d4c25d3/pkg/xds/context/mesh_context_builder.go#L276-L300

```go
case desc.IsDestination:
   destResources = append(destResources, resources.GetItems()
```

And later passed to `NewDestinationIndex(destResources...)`

This leverages existing patterns to make code easier to read and the day we add an extra Destination it works very easily.